apps/web/src/app/(profile)/[username]/_components/dashboard/index.tsx

```diff
@@ -16,7 +16,7 @@ import { Bookmark, CheckCircle, ChevronRightSquare, Play, Settings, Text } from
 import { getRelativeTimeStrict } from '~/utils/relativeTime';
 import { stripProtocolAndWWW } from '~/utils/stringUtils';
 
-import { type BadgeInfo } from './_actions';
```

Comment:
i dont think badges needs it own route. whats the thinking here?

Reply:
I abstracted some generic badges. Will badges be specific in the end? In which case I probably do not need this abstraction layer.

Eg: The endpoint says "Give me a bronze badge with the title 'Super Awesome Badge'"

versus
The endpoint says "Give me the 'Super Awesome Badge Badge'"

Reply:
> I abstracted some generic badges. Will badges be specific in the end? In which case I probably do not need this abstraction layer.
> 
> 
> 
> Eg: The endpoint says "Give me a bronze badge with the title 'Super Awesome Badge'"
> 
> 
> 
> versus
> 
> The endpoint says "Give me the 'Super Awesome Badge Badge'"

oops i thought this was an exposed route. it's not

Reply:
Yeah its not. I wrote some badge logic around bronze/silver/gold/platinum to award the badge more easily from the backend.

Reply:
This code should be removed (the frontend stuff) before merge, just editing getBadges and the creation of badges when necessary.

Reply:
The frontend stuff has been removed now